src/main/java/ua/bookstore/online/repository/book/BookRepository.java

```diff
@@ -1,8 +1,13 @@
 package ua.bookstore.online.repository.book;
 
+import java.util.List;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import ua.bookstore.online.model.Book;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    Optional<Book> findByIsbn(String isbn);
+
+    List<Book> findAllByIdOrIsbn(Long id, String isbn);
```

Comment:
In BookService I already created a separate method to validateIsbnUniqueness. That needed because I meet an issue when I have soft-deleted book: I can't save new one with this ISBN. So I decided to check it in service logic opposite db.